onnx/defs/nn/defs.cc

```diff
@@ -266,6 +267,10 @@ and computes the output.)DOC";
             schema.Attr("output_shape",
                         "The shape of the output.",
                         AttributeProto::INTS, OPTIONAL);
+            schema.Attr("adjs",
+                        "The zero-padding (adjustment) added to one side of the output."
+                        " If output_shape is set, this attribute will be ignored.",
+                        AttributeProto::INTS, OPTIONAL);
```

Comment:
if it's optional, please try to use the API with default value set.

Reply:
To support both output_shape and adjs, ideally a specific attribute verification/parseing logic (If output_shape is set, this attribute will be ignored) should be put in-place. This is something I discussed with Edward on issue #362 . Any thoughts please?

Reply:
The default value should be a list of 0, the length of the list is determined by the input dimension.

Regarding https://github.com/onnx/onnx/issues/362, I am fine with both solutions. My feeling is that extending existing defs/checker should be much easier to implement. I agree with Ed, we should be very clear in (English) description.

Reply:
So, a convention in PyTorch is that if you have a single-element list where an n-element list is expected, we expand the list to cover the number of required elements, replicating the element. However, I'm also pretty OK with keeping things simple for people and NOT defining a default value here; you should just be expected to handle the case when the attribute is missing, which seems pretty reasonable to me. @linkerzhang, is there a reason to put a default value that I am not seeing?

Reply:
I have been thinking of asking default value for all "optional" attributes in operator declaration. It's kind of like initialization when creating a variable. 
The op declaration here is just a standard (having some documentation functionality), so it's better for us to clarify it clearly.
But this case reminds me that we may have default value in the implementation without caller's specification, say if it's a "int" attribute and it's optional, its default value will be "0" if caller does not specify the default value explicitly. In this way, we can ensure that our documentation generated will always have a default value for optional attributes.